source/ShopPanel.cpp

```diff
@@ -616,6 +618,32 @@ bool ShopPanel::Scroll(double dx, double dy)
 
 
 
+void ShopPanel::DoFind(const string &text)
+{
+	int index = FindItem(text);
+	if(index >= 0 && index < static_cast<int>(zones.size()))
+	{
+		vector<Zone>::const_iterator best = std::next(zones.begin(), index);
+		if(best->GetShip())
+			selectedShip = best->GetShip();
+		else
+			selectedOutfit = best->GetOutfit();
+		MainAutoScroll(best);
+	}
+	return;
+}
+
+
+
+int ShopPanel::Search(const string &str, const string &sub)
+{
+	auto it = search(str.begin(), str.end(), sub.begin(), sub.end(),
+		[](char a, char b) { return toupper(a) == toupper(b); });
+	return (it == str.end() ? -1 : it - str.begin());
+}
```

Comment:
Since this is just the same Search function as is used in MapPanel, we should probably make it a static function of some other class. Perhaps put it in Format, which I believe all these panels already have access to.

Reply:
Ahh alright, done